src/cache/file_cache.h

```diff
@@ -101,24 +102,55 @@ class FileCache {
                                   string::CommandLine command_line,
                                   string::Version version);
 
+  static string::HandledHash Hash(string::HandledSource code,
```

Comment:
Why duplicate these methods and not modify existing ones? `extra_files` can be empty vector and we can follow existing logic in this case.